internal/resource/http.go

```diff
@@ -212,10 +213,10 @@ func (c HttpClient) getReaderWithHeader(ctx context.Context, url string, header
 		}
 	}
 
+	ctx, cancelFn := context.WithCancel(context.Background())
 	if c.timeout != 0 {
-		ctxTo, cancel := context.WithTimeout(ctx, c.timeout)
-		ctx = ctxTo
-		defer cancel()
+		cancelFn()
```

Comment:
Is this necesary?

Reply:
I suppose it's not if I had something like 

```
var ctx, cancelFn
if ... {
  ctx = context.WithTimeout
} else
  ctx = context.WithCancel
}
```

but the way it's written the earlier context must be cancelled or it'll leak stuff.

Reply:
Oh gross, is it not gc'd ?

Reply:
I'd argue for the second construction, but not strongly.